scripts/publish-docker-images

```diff
@@ -21,8 +21,8 @@ IMAGE_NAME="amazon/amazon-ecs-agent"
 
 AWS_PROFILE=""
 STAGE_S3_BUCKET=""
-
-source $(dirname "${0}")/publishing-common.sh
+S3_ACL_OVERRIDE=""
+source scripts/publishing-common.sh
```

Comment:
Depending on why this handling is being updated, you may want to consider an alternative to the prior path expansion. In some situations (and some versions of bash) such as sourcing a script: the existing `$(dirname "$0")` will expand to the sourcing script's `$0` and return the wrong path for sourcing `publishing-common.sh`.

Currently, this makes the assumption about where the script is being called or sourced from and should not be merged as is.

Reply:
Here is a fairly well accepted and robust means to resolve the path for use when you can guarantee a script will be using `bash`:

```
DIR="$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )"
```

---

For example of why this is needed:

This is what happens when you source a script:

```
$ source /tmp/mydir.sh 
# #!/usr/bin/env bash
# from_bash_source="$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )"
# from_dollar_0="$(cd "$(dirname "$0")"; pwd -P )"
# echo using bash_source=$from_bash_source
# echo using \$0=$from_dollar_0
I'm working out of: /tmp/myworkdir
using bash_source=/tmp
using $0=/tmp/myworkdir
```

And this is what happens when you run a script directly:

```
$ /tmp/mydir.sh 
# #!/usr/bin/env bash
# from_bash_source="$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )"
# from_dollar_0="$(cd "$(dirname "$0")"; pwd -P )"
# echo using bash_source=$from_bash_source
# echo using \$0=$from_dollar_0
I'm working out of: /tmp/myworkdir
using bash_source=/tmp
using $0=/tmp
```

Reply:
Got it. Thanks a lot. 👍